### ISSUE TITLE
[RFC] Moving straight.el to GitHub Organization

### ISSUE THREAD

**Issue by raxod502** [radian-software/straight.el#946]
I think it's high time that we move `straight.el` from my personal GitHub account to a GitHub Organization. This has a few advantages:
* Ability for co-authors such as @progfolio to fully manage the repository (e.g., administrative settings, adding other collaborators)
* Access to more GitHub features (many things are gated behind putting a repository in an organization)
* Easy way to grant access for collaborators to multiple related repositories, e.g. [GNU ELPA Mirror](https://github.com/raxod502/gnu-elpa-mirror) along with `straight.el`

I also have a personal interest in this. As some of you may know, in mid-2021, I received spurious legal threats from a former employer who felt personally offended about my open-source work. After consulting with a lawyer, I was told that the standard advice is to conduct open-source work as part of a single-member LLC, which is what I'm doing going forward. The idea is that if you are sued, then only the assets of the LLC can be subject to damages, rather than your personal savings. So, I have a GitHub Organization for the legal entity I've formed: [radian-software](https://github.com/radian-software).

On the bookkeeping side, here is what would change:
* Repository would move from https://github.com/raxod502/straight.el to https://github.com/radian-software/straight.el
* Copyright notices would change from `Copyright (c) 2017 Radon Rosborough` to `Copyright (c) 2017-2022 Radian LLC and contributors`
* Lots of URLs would be updated

On the legal side, here is what would change:
* My personal contributions will be transferred (by a legal contract I have on file) from me to Radian LLC
* Everyone else's contributions remain their own, as before
* `straight.el` remains licensed under the MIT License and available under the same terms as before
* If somebody else decides that my open-source work offends them, I don't have to be as scared of a lawsuit

Now since I had the bright idea to make `straight.el` do everything super dynamically, changing the repository URL actually has the potential to break things. I want to make sure we avoid that. I think for the most part things should be pretty smooth, because GitHub automatically redirects the old repository name to the new one (as long as I don't create another repository at the original location, which I won't). So on the technical side:
* I would update all references to `raxod502/straight.el` in the repository to say instead `radian-software/straight.el`
* All previous versions of the bootstrap snippet will continue to function as before, since `raw.githubusercontent.com` handles repository renames transparently
* Git clones of `straight.el` will continue to function as before, both during bootstrap and during subsequent Git operations under all versions of `straight.el`, since Git operations handle repository renames transparently (albeit with a nonfatal warning about updating the remote URL in some cases)
* When people update `straight.el` to a revision that includes the URL update commit, the embedded client-side recipe for `straight.el` will get the new repository path. This will have no visible effect until `M-x straight-normalize-package` (or a dependent command) is run, at which point `straight.el` will offer to update the repository URL automatically
* To be safe, I'll cut a new lockfile version schema and bootstrap snippet version to go along with the URL update, and afterwards will do tests with a variety of client situations to verify there isn't any interruption of existing workflows

In conclusion, I think this change should have fairly minimal impact, both from a semantic perspective and from a technical one. But since `straight.el` is a critical piece of many people's daily workflows with Emacs, I do want to put this out there for community review and comment before taking any action, and want to give people a chance to point out any problems that haven't occurred to me.

Definitely want to get @progfolio's thoughts here as well, since he's frankly doing a lot more work than I am on `straight.el` these days and is responsible for most of the big improvements in the last year or two.

**Comment by progfolio** [comment]
Sounds like a good plan of action to me.

**Comment by raxod502** [comment]
Migration is complete, `straight.el` appears to still bootstrap correctly on both master and develop, and CI is working, but please report any issues that might be caused by this and I'll get those sorted promptly.

**Comment by theophilusx** [comment]
Probably need to update the example bootstrap code in the README as it still references your personal github repo in the URL. This is causing straight to popup a menu to rename the origin. Can do a PR if you want, but suspect it is such a trivial edit, it would be just as easy to edit and push as merge a PR?

**Comment by progfolio** [comment]
@theophilusx The README should be updated on the "develop" branch. Once we merge this change to the "master" branch it will be reflected on the default github page for the repo.

**Comment by raxod502** [comment]
Indeed, master should still be using the old repo name while develop should use the new one. Either way should be stable and bug-free.

**Comment by progfolio** [comment]
You should be able to hit `d` to have straight rename the origin to the new one. Have you tried that option?

**Comment by reuleaux** [comment]
Thanks a lot, and having tried this now (hitting `d`) this works indeed. - It would be nice however, if these leftovers from
the GitHub organization were not necessary any more in the future. - Thanks again. - A.

**Comment by raxod502** [comment]
This should only be a one-time operation - but it's worth noting that you'll encounter this prompt and required action any time one of your packages migrates to a different GitHub repository or organization name. So it's worth considering whether `straight.el` should be made to transparently fix those (I think it likely should), but it won't be a special case related to this issue.

**Comment by danielpza** [comment]
I recently upgraded all `straight-pull-all` and when trying to rebulid the packages I encountered an error when building editorconfig package:

```
Debugger entered--Lisp error: (void-variable root)
  eval(root)
  straight--activate-package-autoloads((:type git :flavor melpa :host github :repo "editorconfig/editorconfig-emacs" :package "editorconfig" :local-repo "editorconfig-emacs"))
  #f(compiled-function () #<bytecode -0x15c018de404766a9>)()
  straight--transaction-exec(use-package-\"fd7eeda9dbc814ea9d6411fedf604a84\"-nil-nil :now #f(compiled-function () #<bytecode -0x15c018de404766a9>))
  straight-use-package(editorconfig)
```

my config:

```
(use-package editorconfig
  :straight t
  :config
  (editorconfig-mode))
```

After disabling the editorconfig package the rest of the packages build correctly, so it's only happening for editorconfig so far.

I'm not sure if this is a straight or editorconfig issue.

**Comment by progfolio** [comment]
Could you share the contents of straight's build-cache.el file?
Should be located at EMACS-USER-DIR/straight/build-cache.el.

**Comment by danielpza** [comment]
https://gist.github.com/danielpza/1dfca0af637f988b5cb73b2130d80f87

**Comment by bestlem** [comment]
Looking at the above comment https://github.com/radian-software/straight.el/issues/946#issuecomment-1145504974

I have the same but surely delete the remote and use the current URL does the wrong thing - in that it puts back the raxod502 entry which is the one we want to remove?


In the straight repo .git/config I have
```
[remote "origin"]
	url = https://github.com/raxod502/straight.el.git
	fetch = +refs/heads/*:refs/remotes/origin/*
```

**Comment by JoJoFringe** [comment]
I don't understand any of it and I've tried so hard. I just need my phone
unhacked

**Comment by raxod502** [comment]
@JoJoFringe Hi, it seems like you're in the wrong place. This is the bug tracker for an open-source project that has nothing to do with mobile devices. If you have concerns about your privacy or security on GitHub, please reach out to the GitHub Support team, which is unaffiliated with this project.

**Comment by raxod502** [comment]
@bestlem Can you open a separate issue for this, and include info about what revision of `straight.el` you're using? This should no longer be happening on the most recent version but there could be a bug.

**Comment by JoJoFringe** [comment]
Thank you so much.  For taking time to answer my question
